napari_skimage_regionprops/_table.py

```diff
@@ -89,43 +103,104 @@ def set_content(self, table : dict):
                 self._view.setItem(j, i, QTableWidgetItem(str(value)))
 
     def get_content(self) -> dict:
-        """
-        Returns the current content of the table
-        """
+        """Return the current content of the table."""
         return self._table
 
     def update_content(self):
-        """
-        Read the content of the table from the associated labels_layer and overwrites the current content.
-        """
+        """Overwrite current content with data from associated labels layer."""
         self.set_content(self._layer.properties)
 
+
 @register_function(menu="Measurement > Show table (nsr)")
-def add_table(labels_layer: napari.layers.Layer, viewer:napari.Viewer) -> TableWidget:
-    """
-    Add a table to a viewer and return the table widget. The table will show the `properties` of the given layer.
+def add_table(labels_layer: napari.layers.Layer,
+              viewer: napari.Viewer) -> TableWidget:
     """
+    Add a table to a viewer and return the table widget.
 
+    The table will show the `properties` of the given layer.
+    """
     dock_widget = get_table(labels_layer, viewer)
     if dock_widget is None:
         dock_widget = TableWidget(labels_layer)
         # add widget to napari
-        viewer.window.add_dock_widget(dock_widget, area='right', name="Properties of " + labels_layer.name)
+        viewer.window.add_dock_widget(dock_widget,
+                                      area='right',
+                                      name="Properties of " + labels_layer.name)
     else:
         dock_widget.set_content(labels_layer.properties)
         if not dock_widget.parent().isVisible():
             dock_widget.parent().setVisible(True)
 
     return dock_widget
 
-def get_table(labels_layer: napari.layers.Layer, viewer:napari.Viewer) -> TableWidget:
+
+def append_table(table_widget: QTableWidget, table: dict):
     """
-    Searches inside a viewer for a given table and returns it. If it cannot find it,
-    it will return None.
+    Append data to a currently active table widget in the viewer.
+
+    The function finds the table widget in the `viewer`, the name of which
+    matches `name`. The data in `table` is then appended to this table and
+    displayed. If the provided data to be added to the table widget fails to
+    provide some of the fields, the field is filled with 'NaN'
+
+    Parameters
+    ----------
+    table_widget : QTableWidget
+        Table widget instance in the viewer.
+    table : dict
+        Data to be appended to the table widget.
+
+    Returns
+    -------
+    None.
+
     """
+    _table = table_widget._table
+
+    for key in _table.keys():
+        if key in table.keys():
+
+            _table[key] += list(table[key])
+        else:
+            _table[key] += list('NaN')
```

Comment:
I think if one merges a table with columns A,B, C and a table with B, C, D, either A or D is lost. I suggest using pandas for merging tables. 
Documentation: https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.append.html
Example: https://github.com/haesleinhuepf/scripts/blob/master/append_tables.ipynb

Reply:
Yes, I think the current implementation will only keep columns that intersect with the ones present in the table. I was also thinking of sing pandas as table backend - should I add this to this PR or rather as a separate one?